Shared/Samples/Change viewpoint/ChangeViewpointView.swift

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct ChangeViewpointView: View {
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    /// The visible area on the map.
+    @State private var visibleArea: ArcGIS.Polygon?
+    
+    /// The scale of the map view.
+    @State private var scale: Double = 0
+    
+    /// The current viewpoint type.
+    @State private var viewpointType: ViewpointType = .centerAndScale
+    
+    enum ViewpointType: CaseIterable {
+        case geometry, centerAndScale, animate
+        
+        /// A human-readable label for the viewpoint type.
+        var label: String {
+            switch self {
+            case .geometry: return "Geometry"
+            case .centerAndScale: return "Center & Scale"
+            case .animate: return "Animate"
+            }
+        }
+    }
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            VStack {
+                MapView(map: model.map, graphicsOverlays: [model.graphicsOverlay])
+                    .onScaleChanged { scale = $0 }
+                    .onVisibleAreaChanged { visibleArea = $0 }
+                    .task(id: viewpointType) {
+                        switch viewpointType {
+                        case .centerAndScale:
+                            do {
+                                try await mapViewProxy.setViewpointCenter(model.londonEast, scale: 4e4)
+                            } catch {}
```

Comment:
I asked about why it throws in the channel. It doesn't need to block this PR though.

```suggestion
                            try? await mapViewProxy.setViewpointCenter(model.londonEast, scale: 4e4)
```

You'll see the warning below. That is a Swift issue https://github.com/apple/swift/issues/45672

> Result of 'try?' is unused

Reply:
I used `_ =` to silence the warning, is there a better way to do so?

Reply:
Underscore is all good. When Apple fixes this bug, I'll clean through the project.